osu.Game.Rulesets.Catch/UI/Catcher.cs

```diff
@@ -78,17 +79,17 @@ public class Catcher : SkinReloadableDrawable, IKeyBindingHandler<CatchAction>
         /// </summary>
         private readonly Container<CaughtObject> droppedObjectTarget;
 
-        public CatcherAnimationState CurrentState { get; private set; }
+        [Cached]
+        protected readonly Bindable<CatcherAnimationState> CurrentStateBindable = new Bindable<CatcherAnimationState>();
+
+        public CatcherAnimationState CurrentState => CurrentStateBindable.Value;
 
         /// <summary>
         /// The width of the catcher which can receive fruit. Equivalent to "catchMargin" in osu-stable.
         /// </summary>
         public const float ALLOWED_CATCH_RANGE = 0.8f;
 
-        /// <summary>
-        /// The drawable catcher for <see cref="CurrentState"/>.
-        /// </summary>
-        internal Drawable CurrentDrawableCatcher => currentCatcher.Drawable;
+        internal Texture CurrentTexture => ((ICatcherSprite)currentCatcher.Drawable).CurrentTexture;
```

Comment:
The fact that this exists at all aside, is there a need for this to be `internal` rather than `private`?

Reply:
"Catcher trail" sprites are generated while the catcher is dashing. The trail sprite uses a snapshot of the catcher sprite (the `CurrentTexture`), and trail sprites shouldn't be animated when the catcher sprite is animated.
Currently, the logic of trail sprite generation is in the separate `CatcherTrailDisplay` class and accessing properties of the catcher including `CurrentTexture`, thus `internal`. I find this is a weird way but modifying this logic is not the scope of this PR.

Reply:
Can we replace the `this` passed in to `CatcherTrailDisplay` with a `Func<Texture>`?

Reply:
No, `CatcherTrailDisplay` is getting `Position` and `Scale` of the catcher in addition to the current texture.

Reply:
I guess we can leave that for another day then. I'm sure we can come up with something better.